File: backend/LeParc.ResidentialFinance.Application/Interfaces/Services/ReportService.cs
using LeParc.ResidentialFinance.Application.DTOs.Reports;
using LeParc.ResidentialFinance.Application.Interfaces.Repositories;
using LeParc.ResidentialFinance.Application.Interfaces.Services;

namespace LeParc.ResidentialFinance.Application.Services;

/// <summary>
/// Implementa os casos de uso relacionados aos relatórios financeiros.
/// </summary>
public sealed class ReportService : IReportService
{
    private readonly IReportRepository _reportRepository;

    /// <summary>
    /// Inicializa uma nova instância do serviço de relatórios.
    /// </summary>
    public ReportService(IReportRepository reportRepository)
    {
        _reportRepository = reportRepository;
    }

    /// <summary>
    /// Obtém o resumo geral do sistema financeiro.
    /// </summary>
    public async Task<FinancialSummaryResponse> GetFinancialSummaryAsync(
        CancellationToken cancellationToken = default)
    {
        var totals = await _reportRepository.GetFinancialTotalsAsync(
            cancellationToken);

        var balance = totals.TotalIncome - totals.TotalExpense;

        return new FinancialSummaryResponse(
            TotalPeople: totals.TotalPeople,
            TotalTransactions: totals.TotalTransactions,
            TotalIncome: totals.TotalIncome,
            TotalExpense: totals.TotalExpense,
            Balance: balance);
    }
}

File: frontend/src/services/report.service.ts
import { api } from "../api/axios";
import type { FinancialSummary } from "../types/report.types";

const REPORT_SUMMARY_ENDPOINT = "/reports/summary";

/**
 * Centraliza as operações relacionadas aos relatórios financeiros.
 */
export const reportService = {
  /**
   * Obtém o resumo financeiro geral do sistema.
   */
  async getSummary(): Promise<FinancialSummary> {
    const response = await api.get<FinancialSummary>(
      REPORT_SUMMARY_ENDPOINT,
    );

    return response.data;
  },
};